Packages/com.unity.inputsystem/InputSystem/API/Input.cs

```diff
@@ -0,0 +1,1862 @@
+
+                        maxValue = Mathf.Max(maxValue,
+                            Mathf.Clamp01(GetMouseButtonControl(mouse, input).ReadValue()));
+                    }
+
+                    break;
+                case InputDeviceType.Gamepad:
+                    for (var i = 0; i < s_Gamepads.Length; i++)
+                    {
+                        if (slot != InputSlot.All && (int)slot != i) continue;
+
+                        var gamepad = s_Gamepads[i];
+                        if (gamepad == null) continue;
+
+                        maxValue = Mathf.Max(maxValue,
+                            Mathf.Clamp01(GetGamepadButtonControl(gamepad, input).ReadValue()));
+                    }
+
+                    break;
+                case InputDeviceType.Joystick:
+                    for (var i = 0; i < (int)InputSlot.Joystick_Max; i++)
+                    {
+                        if (slot != InputSlot.All && (int)slot != i) continue;
+
+                        var joystick = s_Joysticks[i];
+                        if (joystick == null) continue;
+
+                        var control = GetJoystickButtonControl(joystick, input);
+                        if (control == null) continue;
+
+                        maxValue = Mathf.Max(maxValue, Mathf.Clamp01(control.ReadValue()));
+                    }
+
+                    break;
+                case InputDeviceType.Invalid:
+                default:
+                    break;
+            }
+
+            return maxValue;
+        }
+
+        /// <summary>
+        /// Turns any two inputs into an axis value between -1 and 1.
+        /// </summary>
+        /// <param name="negativeAxis">Control from Inputs enum.</param>
+        /// <param name="positiveAxis">Control from Inputs enum.</param>
+        /// <returns>Value from [-1, 1] inclusive. If both controls are fully actuated, 0 will be returned.</returns>
+        public static float GetAxis(Inputs negativeAxis, Inputs positiveAxis)
+        {
+            return GetAxis(positiveAxis) - GetAxis(negativeAxis);
+        }
+
+        internal static Vector2 NormalizeAxis(Vector2 axis, float deadzone)
+        {
+            var currentMag = axis.magnitude;
+            if (currentMag < deadzone)
+                return Vector2.zero;
+            var newMag = Mathf.InverseLerp(deadzone, 1.0f, currentMag);
+            return axis * (newMag / currentMag);
+        }
+
+        /// <summary>
+        /// Turns any four inputs into a normalized vector.
+        /// </summary>
+        /// <param name="left"></param>
+        /// <param name="right"></param>
+        /// <param name="up"></param>
+        /// <param name="down"></param>
+        /// <returns></returns>
+        public static Vector2 GetAxis(Inputs left, Inputs right, Inputs up, Inputs down)
+        {
+            var axis = GetAxisRaw(left, right, up, down);
+            return NormalizeAxis(axis, InputSystem.settings.defaultDeadzoneMin);
```

Comment:
Suggest we pass optional deadzone into this function and use the default if non passed.